src/pages/ListTableList/components/UpdateForm.tsx

```diff
@@ -0,0 +1,239 @@
+import { Button, DatePicker, Form, Input, Modal, Radio, Select, Steps } from 'antd';
+import React, { Component } from 'react';
+
+import { FormComponentProps } from 'antd/es/form';
+import { TableListItem } from '../data.d';
+
+export interface FormValueType extends Partial<TableListItem> {
+  target?: string;
+  template?: string;
+  type?: string;
+  time?: string;
+  frequency?: string;
+}
+
+export interface UpdateFormProps extends FormComponentProps {
+  onCancel: (flag?: boolean, formVals?: FormValueType) => void;
+  onSubmit: (values: FormValueType) => void;
+  updateModalVisible: boolean;
+  values: Partial<TableListItem>;
+}
+const FormItem = Form.Item;
+const { Step } = Steps;
+const { TextArea } = Input;
+const { Option } = Select;
+const RadioGroup = Radio.Group;
+
+export interface UpdateFormState {
+  formVals: FormValueType;
+  currentStep: number;
+}
+
+class UpdateForm extends Component<UpdateFormProps, UpdateFormState> {
+  static defaultProps = {
+    handleUpdate: () => {},
+    handleUpdateModalVisible: () => {},
+    values: {},
+  };
+
+  formLayout = {
+    labelCol: { span: 7 },
+    wrapperCol: { span: 13 },
+  };
+
+  constructor(props: UpdateFormProps) {
+    super(props);
+
+    this.state = {
+      formVals: {
+        name: props.values.name,
+        desc: props.values.desc,
+        key: props.values.key,
+        target: '0',
+        template: '0',
+        type: '1',
+        time: '',
+        frequency: 'month',
+      },
+      currentStep: 0,
+    };
+  }
+
+  handleNext = (currentStep: number) => {
+    const { form, onSubmit: handleUpdate } = this.props;
+    const { formVals: oldValue } = this.state;
+    form.validateFields((err, fieldsValue) => {
+      if (err) return;
+      const formVals = { ...oldValue, ...fieldsValue };
+      this.setState(
+        {
+          formVals,
+        },
+        () => {
+          if (currentStep < 2) {
+            this.forward();
+          } else {
+            handleUpdate(formVals);
+          }
+        },
+      );
+    });
+  };
+
+  backward = () => {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'backward' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4665582)